bvm/ballerina-core/src/main/java/org/ballerinalang/util/transactions/TransactionResourceManager.java

```diff
@@ -254,30 +308,69 @@ void endXATransaction(String transactionId, int transactionBlockId) {
     void rollbackTransaction(String transactionId, int transactionBlockId) {
         endXATransaction(transactionId, transactionBlockId);
         notifyAbort(transactionId, transactionBlockId, true);
+        participantRegistry.participantFailed(transactionId);
     }
 
-    private void removeContextsFromRegistry(String transactionCombinedId) {
+    private void removeContextsFromRegistry(String transactionCombinedId, String gTransactionId) {
         resourceRegistry.remove(transactionCombinedId);
         xidRegistry.remove(transactionCombinedId);
+        participantRegistry.remove(gTransactionId);
     }
 
     private String generateCombinedTransactionId(String transactionId, int transactionBlockId) {
         return transactionId + ":" + transactionBlockId;
     }
 
-    private void invokeCommittedFunction (String transactionId, int transactionBlockId) {
+    private void invokeCommittedFunction(String transactionId, int transactionBlockId) {
         BFunctionPointer fp = committedFuncRegistry.get(transactionBlockId);
+        BValue[] args = {new BString(transactionId + ":" + transactionBlockId)};
         if (fp != null) {
-            BValue[] args = { new BString(transactionId + ":" + transactionBlockId)};
             BVMExecutor.executeFunction(fp.value().getPackageInfo().getProgramFile(), fp.value(), args);
         }
+        List<BFunctionPointer> funcs = participantRegistry.getCommittedFuncs(transactionId);
+        invokeFunctions(args, funcs);
+        participantRegistry.purge(transactionId);
+
     }
 
-    private void invokeAbortedFunction (String transactionId, int transactionBlockId) {
+    private void invokeAbortedFunction(String transactionId, int transactionBlockId) {
```

Comment:
Do we need this function?

Reply:
I left it because when we clarify the abort/rollback behavior from the coordinator we going to need this.